pilot/pkg/xds/gen2.go

```diff
@@ -15,16 +15,34 @@
 package xds
 
 import (
+	"encoding/json"
 	"time"
 
+	corev3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
 	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
 	"google.golang.org/grpc/codes"
 
 	"istio.io/istio/pilot/pkg/model"
+	"istio.io/pkg/env"
+	istioversion "istio.io/pkg/version"
 )
 
 // gen2 provides experimental support for extended generation mechanism.
 
+// IstioControlPlaneInstance defines the format Istio uses for when creating Envoy config.core.v3.ControlPlane.identifier
+type IstioControlPlaneInstance struct {
+	// The Istio Pilot component type (e.g. "pilot")
+	Component string
+	// The ID of the component instance
```

Comment:
More info please. 
- network/IP - since IP by itself is not enough in multi-network cases
- cluster/namespace/pod - istiod can be in multiple clusters - but not sure how it'll work if istiod is on VMs